dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/hibernate/HibernateProgramStore.java

```diff
@@ -105,4 +106,14 @@ public HibernateProgramStore( SessionFactory sessionFactory, JdbcTemplate jdbcTe
 
         return getQuery( hql ).setParameter( "dataEntryForm", dataEntryForm ).list();
     }
+
+    public boolean hasOrgUnit( Program program, OrganisationUnit organisationUnit )
+    {
+        NativeQuery<Long> query = getSession().createNativeQuery(
+            "select programid from program_organisationunits where programid = :pid and organisationunitid = :ouid" );
+        query.setParameter( "pid", program.getId() );
+        query.setParameter( "ouid", organisationUnit.getId() );
+
+        return query.getResultList().size() == 1;
```

Comment:
Maybe slightly safer to do `!query.getResultList().isEmpty()`.

Reply:
Changed.